src/main/java/org/candlepin/resource/ProductResource.java

```diff
@@ -99,7 +99,11 @@ private Product fetchProduct(String productUuid) {
 
     @Override
     public Iterable<ProductDTO> getProducts() {
-        return this.translator.translateQuery(this.productCurator.listAll(), ProductDTO.class);
+        List<Product> products = this.productCurator.listAll();
+        return products.stream().map(this.translator.getStreamMapper(
```

Comment:
Same deal here. We can avoid the collection step by just returning a stream. If we are not able to change that for OpenAPI reasons (I don't see why we wouldn't be able to, the resulting response type is still "list"), then we can at least make it clearer by stacking the operations at the same indentation level:

```
return this.productCurator.listAll()
  .stream()
  .map(this.translator.getStreamMapper(Product.class, ProductDTO.class))
  .toList();
```

Note this occurs in a number of places throughout this PR.